haystack/components/generators/hugging_face_local.py

```diff
@@ -84,37 +83,21 @@ def __init__(
         """
         transformers_import.check()
 
-        huggingface_pipeline_kwargs = huggingface_pipeline_kwargs or {}
+        self.token = token
         generation_kwargs = generation_kwargs or {}
 
-        self.token = token
-        token = token.resolve_value() if token else None
-
-        # check if the huggingface_pipeline_kwargs contain the essential parameters
-        # otherwise, populate them with values from other init parameters
-        huggingface_pipeline_kwargs.setdefault("model", model)
-        huggingface_pipeline_kwargs.setdefault("token", token)
-
-        device = ComponentDevice.resolve_device(device)
-        device.update_hf_kwargs(huggingface_pipeline_kwargs, overwrite=False)
-
-        # task identification and validation
-        if task is None:
-            if "task" in huggingface_pipeline_kwargs:
-                task = huggingface_pipeline_kwargs["task"]
-            elif isinstance(huggingface_pipeline_kwargs["model"], str):
-                task = model_info(
-                    huggingface_pipeline_kwargs["model"], token=huggingface_pipeline_kwargs["token"]
-                ).pipeline_tag
-
-        if task not in SUPPORTED_TASKS:
-            raise ValueError(
-                f"Task '{task}' is not supported. " f"The supported tasks are: {', '.join(SUPPORTED_TASKS)}."
-            )
-        huggingface_pipeline_kwargs["task"] = task
+        huggingface_pipeline_kwargs = resolve_hf_pipeline_kwargs(
```

Comment:
@sjrl why not do this in a separate PR?

Reply:
This is the first PR that needs the abstraction. This only became duplicate code with the addition of another component that uses a HF pipeline.

Reply:
So to be clear I need this function in the newly added `TransformersZeroShotTextRouter` as well. So I opted to put the common code into a new function rather than copy it.

Reply:
Ok np, I was surprised by it, that's all. And wanted to ping @anakin87 for a nod?

Reply:
Looks good. We can iterate on it in the future...